packages/reactotron-app/App/Dialogs/FilterTimelineDialog.js

```diff
@@ -0,0 +1,246 @@
+      },
+      {
+        value: 'benchmark.report',
+        text: 'Benchmark'
+      },
+      {
+        value: 'log',
+        text: 'Log Messages'
+      },
+      {
+        value: 'image',
+        text: 'Images'
+      },
+      {
+        value: 'display',
+        text: 'Display'
+      }
+    ]
+  },
+  {
+    name: 'API',
+    items: [
+      {
+        value: 'api.response',
+        text: 'API Responses'
+      }
+    ]
+  }
+  {
+    name: 'Redux',
+    items: [
+      {
+        value: 'state.action.complete',
+        text: 'Action'
+      },
+      {
+        value: 'saga.task.complete',
+        text: 'Saga'
+      },
+      {
+        value: 'state.values.response',
+        text: 'State Values'
+      },
+      {
+        value: 'state.values.response',
+        text: 'State Keys'
+      },
+      {
+        value: 'state.values.change',
+        text: 'State Values Change'
+      }
+    ]
+  }
+]
+
+const DIALOG_TITLE = 'Filter'
+
+const Styles = {
+  dialog: {
+    borderRadius: 4,
+    padding: 4,
+    width: 450,
+    backgroundColor: Colors.background,
+    color: Colors.foreground
+  },
+  examples: {
+  },
+  example: {
+    padding: 0,
+    margin: '0 0 0 40px',
+    color: Colors.bold
+  },
+  container: {
+    ...AppStyles.Layout.vbox
+  },
+  keystrokes: {
+    ...AppStyles.Layout.hbox,
+    alignSelf: 'center',
+    paddingTop: 10,
+    paddingBottom: 20,
+    fontSize: 13
+  },
+  hotkey: {
+    padding: '0 10px'
+  },
+  keystroke: {
+    backgroundColor: Colors.backgroundHighlight,
+    color: Colors.foreground,
+    padding: '4px 8px',
+    borderRadius: 4
+  },
+  header: {
+    ...AppStyles.Layout.vbox,
+    padding: '1em 2em 0em'
+  },
+  body: {
+    ...AppStyles.Layout.vbox,
+    padding: '1em 2em 4em'
+  },
+  title: {
+    margin: 0,
+    padding: 0,
+    textAlign: 'left',
+    fontWeight: 'normal',
+    fontSize: 24,
+    color: Colors.heading
+  },
+  subtitle: {
+    color: Colors.foreground,
+    textAlign: 'left',
+    padding: 0,
+    margin: 0
+  },
+  fieldLabel: {
+    color: Colors.heading,
+    fontSize: 13,
+    textTransform: 'uppercase'
+  },
+  textField: {
+    borderTop: 0,
+    borderLeft: 0,
+    borderRight: 0,
+    borderBottom: `1px solid ${Colors.line}`,
+    fontSize: 23,
+    color: Colors.foregroundLight,
+    lineHeight: '40px',
+    backgroundColor: 'inherit'
+  }
+}
+
+const INSTRUCTIONS = <div>
+  <p>Enter a path you would like to subscribe.  Here are some examples to get you started:</p>
+  <p style={Styles.example}>user.firstName</p>
+  <p style={Styles.example}>repo</p>
+  <p style={Styles.example}>repo.*</p>
+</div>
+
+@inject('session')
+@observer
+class FilterTimelineDialog extends Component {
+
+  componentWillMount() {
+    // I feel like this is a terrible place to do this
+    // but putting here for now to get things to work
+    this.props.session.timelineCommandFilters =
+      [].concat.apply([], FILTER_OPTIONS.map(grp => grp.items.map(itm => itm.value)))
```

Comment:
I personally do not like this but not sure of another way of doing it unless we move the command/groups out of here or we duplicate the list yet again (it is already duplicated between here and the commands switch statement)